Readme suggestions (#1126)
zooniverse/front-end-monorepo: README.md

Apps should have their directory names prefixed with `app-`, e.g. `/project` becomes `/app-project`.

> ## Deployment

Reviewer: The table of contents needs updating to link to this heading.

> Deploys to production and staging are handled by [Jenkins](https://jenkins.zooniverse.org/job/Zooniverse%20GitHub/job/front-end-monorepo/). Firstly, a base Docker image (`zooniverse/front-end-monorepo`) is built with the top-level Dockerfile, which installs and builds production versions of the `lib-` packages. 

Reviewer: The description of the base image can move down to the list of Docker images.

> Deploys to production and staging are handled by [Jenkins](https://jenkins.zooniverse.org/job/Zooniverse%20GitHub/job/front-end-monorepo/) using Docker images (descriptions found below). 

Reviewer: 'Docker images' should be a link to the Docker images heading.

Reviewer: I gave a brief step-by-step description of the Jenkins job in #1123. It's all been removed here but should we keep it in? I thought it was useful to clarify in my mind what Jenkins does when it's deploying the apps.

Author: I copy and pasted those lines into the docker images section.

Merges to master deploy to a staging Kubernetes instance that uses Panoptes production. This is used for manual end to end behavior testing for new code and design reviews. `https://frontend.preview.zooniverse.org/projects/:project-owner/:project-name/` proxy redirects to the new NextJS app while the rest of sub-domain redirects to PFE. Staging projects can be loaded by adding this query param to the URL: `?env=staging`.

> Deployments to a production Kubernetes instance are triggered by committing a `production-release` git tag on master. This can either be done using the git CLI or using the lita deploy command on slack. `https://www.zooniverse.org/projects/:project-owner/:project-name/classify` proxy redirects to the new NextJS app while the rest of the domain redirects to PFE. Currently the only project that is configured to do this is Planet Hunters TESS. Eventually more projects will migrate when they migrate to the new classifier.

Reviewer: ```suggestion
Commiting the `production-release` tag to master deploys to a production Kubernetes instance. This can either be done using the git CLI or using the lita deploy command on slack. `https://www.zooniverse.org/projects/:project-owner/:project-name/classify` proxy redirects to the new NextJS app while the rest of the domain redirects to PFE. Currently the only project that is configured to do this is Planet Hunters TESS. Eventually more projects will migrate when they migrate to the new classifier.
```
Changed to match the sense of the section about staging deploys.

Reviewer: >using the git CLI or using the lita deploy command

Should we be explicit about how to do this? How much knowledge are we assuming about the person using these instructions?

Author: I'd prefer precise step by step instructions in a more secure place like the org's stackoverflow or how-to-zooniverse. The process on how this works at this point is the same as Panoptes, Caesar, etc. I'm not sure if we should make this public.

More information is available in [ADR 12](docs/arch/adr-12.md) and [ADR 17](docs/arch/adr-17.md)

### Environment variables

> - `PANOPTES_ENV`: sets which Panoptes API endpoint to use. Setting to `production` will use `https://www.zooniverse.org/api` and `staging` to `https://panoptes-staging.zooniverse.org/api`. The yarn build scripts for libraries default to production and overrides the environment to be production for the apps. 

Reviewer: ```suggestion
- `PANOPTES_ENV`: sets which Panoptes API endpoint to use. 
  - `production` will use `https://www.zooniverse.org/api` 
  - `staging` will use `https://panoptes-staging.zooniverse.org/api`.

The yarn build scripts default to production for libraries if `PANOPTES_ENV` is not specified. The apps are always built to the production API.
```

- `NODE_ENV`: sets the environment for library builds.

### Docker images

> - `zooniverse/front-end-monorepo`: Used as a base image to build `zooniverse/fe-project` and `zooniverse/fe-content-pages` images from the Dockerfiles in `packages/app-project` and `packages/app-content-pages`

Reviewer: Built from the Dockerfile in the root directory. It runs `yarn install` and builds the `lib-` packages and static (production) apps.

Reviewer: #1109 moves the static builds back into the Dockerfiles for the respective NextJS apps.

> - `zooniverse/fe-content-pages`: Runs production (static) builds of the NextJS app and are then deployed to Kubernetes

Reviewer: Built from the Dockerfile in `packages/app-content-pages`.

> - `zooniverse/fe-project`: Runs production (static) builds of the NextJS app and are then deployed to Kubernetes

Reviewer: Built from the Dockerfile in `packages/app-project`.

> - `zooniverse/front-end-monorepo`: Used as a base image to build `zooniverse/fe-project` and `zooniverse/fe-content-pages` images from the Dockerfiles in `packages/app-project` and `packages/app-content-pages`. Built from the Dockerfile in the root directory. It runs `yarn install` and builds the `lib-` packages and static (production) apps.

Reviewer: ```suggestion
- `zooniverse/front-end-monorepo`: Built from the Dockerfile in the root directory. It runs `yarn install` and `yarn build` for the `lib-` packages and static (production) apps.
```

> - `zooniverse/fe-content-pages`: Runs production (static) builds of the NextJS app and are then deployed to Kubernetes. Built from the Dockerfile in `packages/app-content-pages`. Runs `yarn start`.

Reviewer: ```suggestion
- `zooniverse/fe-content-pages`: Built from the Dockerfile in `packages/app-content-pages`. Runs `yarn start` in `packages/app-content-pages` from `zooniverse/front-end-monorepo:latest`.
```

> - `zooniverse/fe-project`: Runs production (static) builds of the NextJS app and are then deployed to Kubernetes. Built from the Dockerfile in `packages/app-project`. Runs `yarn start`.

Reviewer: ```suggestion
- `zooniverse/fe-project`: Built from the Dockerfile in `packages/app-project`. Runs `yarn start` in `packages/app-project` from `zooniverse/front-end-monorepo:latest`.
```
